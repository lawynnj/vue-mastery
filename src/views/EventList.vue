<template>
  <div class="event-list">
    <h1>List component</h1>
    <EventCard v-for="event in event.events" :key="event.id" :event="event" />
    <template v-if="page != 1">
      <router-link
        :to="{ name: 'event-list', query: { page: page - 1 } }"
        rel="prev"
        >Prev Page</router-link
      >
      |
    </template>
    <template v-if="hasNextPage">
      <router-link :to="{ name: 'event-list', query: { page: page + 1 } }"
        >Next Page</router-link
      >
    </template>
  </div>
</template>

<script>
import EventCard from '@/components/EventCard.vue'
import { mapActions, mapState } from 'vuex'

export default {
  components: {
    EventCard
  },
  computed: {
    page() {
      return parseInt(this.$route.query.page) || 1
    },
    hasNextPage() {
      return this.event.eventsTotal > this.page * this.event.perPage
    },
    // connect to Vuex state
    ...mapState(['event', 'user'])
  },
  created() {
    const config = {
      page: this.page
    }
    // call a function that will make an API call and update the state
    this.$store.dispatch('event/fetchEvents', config)
  },
  methods: {
    showNotification(notification) {
      this.add(notification)
    }
  },
  actions: {
    ...mapActions('notification', ['add', 'delete'])
  }
}
</script>
<style>
.event-list {
  border: solid green 2px;
}
</style>
