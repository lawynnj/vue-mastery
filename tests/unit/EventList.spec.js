import Vuex from 'vuex'
import EventList from '@/views/EventList.vue'
import { shallowMount, createLocalVue } from '@vue/test-utils'

const localVue = createLocalVue()

localVue.use(Vuex)

describe('EventList', () => {
  let store
  const $route = {
    query: { page: '1' }
  }
  beforeEach(() => {
    const user = {}
    const event = {
      namespaced: true,
      actions: {
        fetchEvents: jest.fn()
      },
      state: {
        perPage: 1,
        events: [],
        eventsTotal: 1
      }
    }
    const notification = {
      actions: {
        add: jest.fn(),
        delete: jest.fn()
      }
    }

    store = new Vuex.Store({
      modules: {
        user,
        event,
        notification
      },
      state: {
        categories: []
      }
    })
  })

  test('EventList renders correctly', () => {
    const wrapper = shallowMount(EventList, {
      store,
      localVue,
      mocks: { $route }
    })
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
