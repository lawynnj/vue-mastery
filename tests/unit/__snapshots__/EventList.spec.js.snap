// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventList EventList renders correctly 1`] = `
<div
  class="event-list"
>
  <h1>
    List component
  </h1>
    
  <!---->
   
  <!---->
</div>
`;
